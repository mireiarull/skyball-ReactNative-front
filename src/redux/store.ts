import {
  configureStore,
  type ThunkAction,
  type Action,
} from "@reduxjs/toolkit";
import { gamesReducer } from "./features/gamesSlice/gamesSlice";
import { uiReducer } from "./features/uiSlice/uiSlice";
import { userReducer } from "./features/userSlice/userSlice";

export const store = configureStore({
  reducer: {
    ui: uiReducer,
    user: userReducer,
    games: gamesReducer,
  },
});

export type AppDispatch = typeof store.dispatch;
export type RootState = ReturnType<typeof store.getState>;
export type AppThunk<ReturnType = void> = ThunkAction<
  ReturnType,
  RootState,
  unknown,
  Action<string>
>;
