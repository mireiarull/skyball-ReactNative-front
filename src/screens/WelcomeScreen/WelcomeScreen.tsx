/* eslint-disable @typescript-eslint/no-unsafe-call */
/* eslint-disable @typescript-eslint/no-unsafe-assignment */
import React from "react";
import {
  View,
  SafeAreaView,
  KeyboardAvoidingView,
  ImageBackground,
  Text,
  Image,
} from "react-native";
import backgroundImage from "./../../../assets/background-image.jpg";
import logo from "./../../../assets/skyball-logo-white.png";
import { TouchableOpacity } from "react-native-gesture-handler";
import buttonStyles from "../../styles/buttonStyles";
import welcomeScreenStyles from "./WelcomeScreenStyles";
import { useNavigation } from "@react-navigation/native";
import { type LoginScreenNavigationProp } from "../../types/navigation.types";
import RoutesEnum from "../../navigation/routes";

const WelcomeScreen = () => {
  const navigation = useNavigation<LoginScreenNavigationProp>();

  return (
    <KeyboardAvoidingView behavior="padding" enabled={true}>
      <ImageBackground
        source={backgroundImage}
        resizeMode="cover"
        style={welcomeScreenStyles.image}
        testID="backgroundImage"
      ></ImageBackground>
      <SafeAreaView>
        <View style={welcomeScreenStyles.container}>
          <View style={welcomeScreenStyles.imageContainer}>
            <Image source={logo}></Image>
          </View>

          <TouchableOpacity
            style={welcomeScreenStyles.button}
            testID={"submitButton"}
            onPress={() => {
              navigation.navigate(RoutesEnum.register);
            }}
          >
            <Text style={buttonStyles.buttonText}>Registrarme</Text>
          </TouchableOpacity>
          <TouchableOpacity
            style={welcomeScreenStyles.button}
            testID={"submitButton"}
            onPress={() => {
              navigation.navigate(RoutesEnum.login);
            }}
          >
            <Text style={buttonStyles.buttonText}>Iniciar sesi√≥n</Text>
          </TouchableOpacity>
          <TouchableOpacity>
            <Text style={welcomeScreenStyles.link}>
              Continuar como invitado
            </Text>
          </TouchableOpacity>
        </View>
      </SafeAreaView>
    </KeyboardAvoidingView>
  );
};

export default WelcomeScreen;
