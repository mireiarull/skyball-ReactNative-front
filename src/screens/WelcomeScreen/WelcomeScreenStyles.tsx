import { Dimensions, StyleSheet } from "react-native";
import buttonStyles from "../../styles/buttonStyles";
import colorStyles from "../../styles/colorStyles";

const welcomeScreenStyles = StyleSheet.create({
  container: {
    display: "flex",
    height: Dimensions.get("window").height,
    padding: 25,
    justifyContent: "flex-end",
    paddingBottom: 100,
  },
  imageContainer: {
    display: "flex",
    flex: 1,
    justifyContent: "center",
    alignSelf: "flex-start",
    paddingBottom: 410,
  },

  button: {
    ...buttonStyles.button,
    marginTop: 10,
    marginBottom: 10,
  },
  image: {
    height: Dimensions.get("window").height,
    width: Dimensions.get("window").width,
    justifyContent: "center",
    alignItems: "center",
    flex: 1,
  },
  link: {
    textAlign: "center",
    fontWeight: "600",
    fontSize: 20,
    color: colorStyles.white,
  },
});

export default welcomeScreenStyles;
