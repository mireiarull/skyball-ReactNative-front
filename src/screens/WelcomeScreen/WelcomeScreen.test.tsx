/* eslint-disable @typescript-eslint/no-unsafe-return */
/* eslint-disable @typescript-eslint/no-unsafe-assignment */
import React from "react";
import { render, screen, fireEvent } from "@testing-library/react-native";
import { Provider } from "react-redux";
import { store } from "../../redux/store";
import WelcomeScreen from "./WelcomeScreen";
import { NavigationContainer } from "@react-navigation/native";
import RoutesEnum from "../../navigation/routes";

const mockedNavigate = jest.fn();

jest.mock("@react-navigation/native", () => {
  const actualNav = jest.requireActual("@react-navigation/native");
  return {
    ...actualNav,
    useNavigation: () => ({
      navigate: mockedNavigate,
    }),
  };
});

describe("Given a WelcomeScreen page", () => {
  describe("When it's rendered", () => {
    test("Then it should show a background image and three buttons", async () => {
      const imageId = "backgroundImage";
      const loginButtontext = "Iniciar sesión";
      const registerButtonText = "Registrarme";
      const guestButtonText = "Continuar como invitado";

      render(
        <Provider store={store}>
          <NavigationContainer>
            <WelcomeScreen />
          </NavigationContainer>
        </Provider>
      );

      const displayedImage = screen.queryByTestId(imageId);
      const loginButton = await screen.getByText(loginButtontext);
      const registerButton = await screen.getByText(registerButtonText);
      const guestButton = await screen.getByText(guestButtonText);

      expect(displayedImage).toBeDefined();
      expect(loginButton).toBeDefined();
      expect(registerButton).toBeDefined();
      expect(guestButton).toBeDefined();
    });

    describe("And the user clicks the 'Registrarme' button", () => {
      test("Then the useNavigation should be called with the register page reference", async () => {
        const registerButtonText = "Registrarme";

        render(
          <Provider store={store}>
            <NavigationContainer>
              <WelcomeScreen />
            </NavigationContainer>
          </Provider>
        );

        const registerButton = await screen.getByText(registerButtonText);
        fireEvent(registerButton, "press");

        expect(mockedNavigate).toHaveBeenCalledWith(RoutesEnum.register);
      });
    });

    describe("And the user clicks the 'Iniciar sesión' button", () => {
      test("Then the useNavigation should be called with the login page reference", async () => {
        const loginButtonText = "Iniciar sesión";

        render(
          <Provider store={store}>
            <NavigationContainer>
              <WelcomeScreen />
            </NavigationContainer>
          </Provider>
        );

        const loginButton = await screen.getByText(loginButtonText);
        fireEvent(loginButton, "press");

        expect(mockedNavigate).toHaveBeenCalledWith(RoutesEnum.login);
      });
    });

    describe("And the user clicks the 'Continuar como invitado' button", () => {
      test("Then the useNavigation should be called with the explore page reference", async () => {
        const continueButtonText = "Continuar como invitado";

        render(
          <Provider store={store}>
            <NavigationContainer>
              <WelcomeScreen />
            </NavigationContainer>
          </Provider>
        );

        const loginButton = await screen.getByText(continueButtonText);
        fireEvent(loginButton, "press");

        expect(mockedNavigate).toHaveBeenCalledWith(RoutesEnum.home);
      });
    });
  });
});
