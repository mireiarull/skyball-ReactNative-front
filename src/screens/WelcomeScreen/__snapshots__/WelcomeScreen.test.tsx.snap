// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given a WelcomeScreen page When it's rendered Then it should match the snapshot 1`] = `
Array [
  <View
    accessibilityIgnoresInvertColors={true}
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "height": 1334,
        "justifyContent": "center",
        "width": 750,
      }
    }
  >
    <Image
      resizeMode="cover"
      source={1}
      style={
        Array [
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
          Object {
            "height": 1334,
            "width": 750,
          },
          undefined,
        ]
      }
      testID="backgroundImage"
    />
  </View>,
  <RCTSafeAreaView>
    <View
      style={
        Object {
          "display": "flex",
          "height": 1334,
          "justifyContent": "flex-end",
          "padding": 25,
          "paddingBottom": 100,
        }
      }
    >
      <View
        style={
          Object {
            "alignSelf": "flex-start",
            "display": "flex",
            "flex": 1,
            "justifyContent": "center",
            "paddingBottom": 410,
          }
        }
      >
        <Image
          source={1}
        />
      </View>
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#5265ff",
            "borderRadius": 90,
            "height": 50,
            "justifyContent": "center",
            "marginBottom": 10,
            "marginTop": 10,
            "opacity": 1,
            "padding": 10,
            "width": "100%",
          }
        }
        testID="submitButton"
      >
        <Text
          style={
            Object {
              "color": "#fff",
              "fontSize": 20,
              "fontWeight": "600",
            }
          }
        >
          Registrarme
        </Text>
      </View>
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#5265ff",
            "borderRadius": 90,
            "height": 50,
            "justifyContent": "center",
            "marginBottom": 10,
            "marginTop": 10,
            "opacity": 1,
            "padding": 10,
            "width": "100%",
          }
        }
        testID="submitButton"
      >
        <Text
          style={
            Object {
              "color": "#fff",
              "fontSize": 20,
              "fontWeight": "600",
            }
          }
        >
          Iniciar sesi√≥n
        </Text>
      </View>
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#fff",
              "fontSize": 20,
              "fontWeight": "600",
              "textAlign": "center",
            }
          }
        >
          Continuar como invitado
        </Text>
      </View>
    </View>
  </RCTSafeAreaView>,
]
`;
